plugins/out_s3/s3.c

```diff
@@ -657,9 +688,143 @@ static int cb_s3_init(struct flb_output_instance *ins,
+                         DEFAULT_PARQUET_RECORD_TYPE);
+            ctx->parquet_record_type = \
+                    flb_sds_create(DEFAULT_PARQUET_RECORD_TYPE);
+            if (ctx->parquet_record_type == NULL) {
+                flb_plg_error(ctx->ins, "Failed to create parquet record type");
+                flb_errno();
+                return -1;
+            }
+        }
+        else {
+            if (strncasecmp(tmp, "json", 4) == 0) {
+                tmp = "jsonl"; /* json should be interpreted as jsonl */
+                flb_plg_info(ctx->ins, "parquet.record_type format is %s", tmp);
+            }
+            else if (strncasecmp(tmp, "msgpack", 7) == 0 ||
+                strncasecmp(tmp, "jsonl", 5) == 0) {
+                flb_plg_info(ctx->ins, "parquet.record_type format is %s", tmp);
+            }
+            else {
+                flb_plg_error(ctx->ins, "unknown parquet.record_type format %s", tmp);
+                return -1;
+            }
+            ctx->parquet_record_type = flb_sds_create_len(tmp, strlen(tmp));
+            if (ctx->parquet_record_type == NULL) {
+                flb_plg_error(ctx->ins, "Failed to create parquet record type");
+                flb_errno();
+                return -1;
+            }
+        }
+
+        tmp = flb_output_get_property("parquet.schema_type", ins);
+        if (!tmp) {
+            flb_plg_info(ctx->ins, "parquet.schema_type format is %s",
+                         DEFAULT_PARQUET_SCHEMA_TYPE);
+            ctx->parquet_schema_type = \
+                    flb_sds_create(DEFAULT_PARQUET_SCHEMA_TYPE);
+            if (ctx->parquet_schema_type == NULL) {
+                flb_plg_error(ctx->ins, "Failed to create parquet schema type");
+                flb_errno();
+                return -1;
+            }
+        }
+        else {
+            if (strncasecmp(tmp, "avro", 4) == 0 ||
+                strncasecmp(tmp, "bigquery", 8) == 0) {
+                flb_plg_info(ctx->ins, "parquet.schema_type format is %s", tmp);
+            }
+            else {
+                flb_plg_error(ctx->ins, "unknown parquet.schema_type format %s", tmp);
+                return -1;
+            }
+            ctx->parquet_schema_type = flb_sds_create_len(tmp, strlen(tmp));
+            if (ctx->parquet_schema_type == NULL) {
+                flb_plg_error(ctx->ins, "Failed to create parquet schema type");
+                flb_errno();
+                return -1;
+            }
+        }
+
+        tmp = flb_output_get_property("parquet.schema_file", ins);
+        if (!tmp) {
+            flb_plg_error(ctx->ins, "parquet.schema_file is missing");
+            return -1;
+        }
+        ctx->parquet_schema_file = flb_sds_create_len(tmp, strlen(tmp));
+        if (ctx->parquet_schema_file == NULL) {
+            flb_plg_error(ctx->ins, "Failed to create parquet schema file");
+            flb_errno();
+            return -1;
```

Comment:
As we are exiting control here, I am not seeing cleanup of:
- ctx->parquet_compression
- ctx->parquet_record_type
- ctx->parquet_schema_type